.github/workflows/pull_request.yml

```diff
@@ -1,16 +1,52 @@
-name: Run action - pull request
+name: Build and test on pull request
 on: [pull_request]
 
-# It is not possible to test a pull request using run_action.yml 
-# because of the action uses the deployed docker image from GHCR.
-
 jobs:
-  default-configuration-test-on-pull-request:
+  build-and-test:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    # if: ${{ !github.event.act }}
     steps:
       - uses: actions/checkout@v3
 
-      - run: pip install PyYAML
+      - name: get packages
+        if: ${{ env.ACT }}
+        run: |
+         sudo apt update -qq && sudo DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
+         python3-pip telnet iptables iproute2 bc cpio rsync
+
+      - name: install-buildroot
+        run: git clone --branch 2022.11.2 https://github.com/buildroot/buildroot buildroot
+      - name: build-kernel
+        run: |
+          cd buildroot && \
+          make BR2_EXTERNAL=../br2-external hifive_unleashed_defconfig && \
+          make -j4 && \
+          cd .. && \
+          mkdir images && \
+          for i in hifive-unleashed-a00.dtb fw_payload.elf Image rootfs.cpio; do cp buildroot/output/images/$i images; done
+      - name: compress
+        run: tar cJvf images.tar.xz images
```

Comment:
If these steps repeat, why not put them in scripts? E.g. some projects have a top-level `ci` dir with scripts. These scripts are then simply called from GHA workflows. We can use the `scripts` directory.

Reply:
You're right, it would be better